### PR TITLE
Adding plating for food.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
This is a bit of a fey mood and is pretty anemic/rushed, willing to take input on implementation prior to merge.

- Removes plates, trays and platters from food icons in food.dmi.
- Adds plates/platters/trays that food can be placed on.
- Removes plate and tray trash items.
- Eating food off a plate dirties the plate, which can then be washed to clean it.

## Why and what will this PR improve
Trash plates/trays has always been a bit strange. This adds a slightly more interesting element to food.

## Authorship
Myself.

## Changelog
:cl:
tweak: Plates are no longer part of food when it emerges from the microwave, and must be retrieved from the dinnerware vendor and used to plate food.
/:cl:

### CODE REVIEW

**Comment by out-of-phaze** [comment]
> Trash plates/trays has always been a bit strange.

paper plates and styrofoam trays, come on......

also cool PR will review soon

File: code/modules/reagents/reagent_containers/food.dm [hunk 1]
```diff
@@ -189,6 +210,11 @@
 	qdel(src)
 
 /obj/item/chems/food/Destroy()
+	if(plate)
```
Comment (out-of-phaze) [inline code/modules/reagents/reagent_containers/food.dm]: this is causing some CI fails. i think when we qdel the food the plate should be lost too, and it should instead be dropped in, i don't know, physically_destroyed or eating code instead

**Review by out-of-phaze — DISMISSED** [review]

**Comment by MistakeNot4892** [comment]
> paper plates and styrofoam trays, come on......


We could add like, ten pack paper plates/plastic plates from cargo. Mostly I just dislike how you can eat a sandwich you made and it spawns a dirty plate that you can do nothing with except dump into the trash.

File: code/modules/food/plates/_plate.dm [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	if(is_dirty)
+		var/image/I = image(icon, "[icon_state]-dirty")
+		I.appearance_flags |= RESET_COLOR
+		I.color = is_dirty
+		add_overlay(I)
+
+/obj/item/plate/attackby(obj/item/W, mob/living/user)
+	// Plating food.
+	if(istype(W, /obj/item/chems/food))
+		if(!user.try_unequip(W))
```
Comment (comma) [inline code/modules/food/plates/_plate.dm:53]: Maybe move this bloc into a proc which can be also called by food attackby? Seems like it's pretty much same code as there.

File: mods/content/government/away_sites/icarus/icarus-1.dmm [hunk 3]
```diff
@@ -708,7 +708,7 @@
 /obj/effect/floor_decal/corner/grey/diagonal{
 	dir = 4
 	},
-/obj/item/trash/plate,
+/obj/item/plate,
```
Comment (comma) [inline mods/content/government/away_sites/icarus/icarus-1.dmm:711]: Maybe needs one of those map path update script things?

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**